### DISCUSSION TITLE
cascadeDelete on user->profile

### DISCUSSION THREAD

**Discussion by dxgi** [prisma/prisma#22117] [Q&A] [answered]
How would I make "profile" do cascadeDelete whenever "user" is deleted. Please refer to the prisma schema below.

```prisma
model User {
    id         Int      @id @default(autoincrement())

    username   String   @unique
    password   String

    profile    Profile  @relation(fields: [profileId], references: [id])
    profileId  Int

    updated_at DateTime @updatedAt
    created_at DateTime @default(now())
}

model Profile {
    id           Int           @id @default(autoincrement())
    display_name String
    bio          String?
    picture      String?
    status       ProfileStatus @default(Offline)
    invisible    Boolean       @default(false)

    updated_at   DateTime      @updatedAt
    created_at   DateTime      @default(now())

    User         User[]
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @dxgi 👋 

Thank you for raising this question.

Looking at the schema you shared, you modelled a `Profile` to have many `Users`. Is that correct? From my understanding, the relationship between a `User` and a `Profile` should be one-to-one, meaning that each `User` can have only one `Profile`, and each `Profile` belongs to only one `User`. Also, a `User` can have many `Profiles` as well. I will advise reviewing your schema.

With respect to the [`Cascade Delete`](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions#cascade), Let's say you have this schema, that shows a one-to-one relationship between `User` and `Profile`
```prisma
model User {
  id Int @id @default(autoincrement())

  username String @unique
  password String

  profile Profile?

  updated_at DateTime @updatedAt
  created_at DateTime @default(now())
}

model Profile {
  id           Int     @id @default(autoincrement())
  display_name String
  bio          String?
  picture      String?
  //status       ProfileStatus @default(Offline)
  invisible    Boolean @default(false)

  updated_at DateTime @updatedAt
  created_at DateTime @default(now())
  userId     Int      @unique
  user       User     @relation(fields: [userId], references: [id], onDelete: Cascade)
}
```
Deleting a `User` record will also delete the `Profile`. Note that we are using the referential action `onDelete: Cascade` which is defined in the [@relation](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#relation) attribute and map to the actions on the foreign key constraint in the underlying database.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
